source/validate_id.cpp

```diff
@@ -779,14 +779,21 @@ bool idUsage::isValid<SpvOpLoad>(const spv_instruction_t* inst,
                        << "' is not a pointer.";
     return false;
   }
-  auto type = usedefs_.FindDef(pointer.second.words[1]);
-  assert(type.first);
-  spvCheck(resultType.second.id != type.second.id,
-           DIAG(resultTypeIndex)
-               << "OpLoad Result Type <id> '" << inst->words[resultTypeIndex]
-               << " does not match Pointer <id> '" << pointer.second.id
-               << "'s type.";
-           return false);
+  auto pointerType = usedefs_.FindDef(pointer.second.words[1]);
+  if (!pointerType.first || pointerType.second.opcode != SpvOpTypePointer) {
```

Comment:
As an aside, I'm finding the use of a std::pair return value from usedefs_.FindDef very hard to read.  I'd rather have a bespoke struct with nice field values.